src/rust/src/lib.rs

```diff
@@ -0,0 +1,189 @@
+use pyo3::prelude::*;
+use pyo3::{types::PyString, PyResult, Python};
+
+static NEEDS_SANITIZE: [bool; 256] = {
+    const __: bool = false;
+    const XX: bool = true;
+    [
+        //  1   2   3   4   5   6   7   8   9   A   B   C   D   E   F
+        __, __, __, __, __, __, __, __, __, __, __, __, __, __, __, __, // 0
```

Comment:
This is cool to look at, and sort of visually explains what's going on. But is there a more compact way to do this, such as initializing the whole array to false then setting the five true values?